Modix.Data/Repositories/MessageRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Models.Core;
+
+namespace Modix.Data.Repositories
+{
+    public interface IMessageRepository
```

Comment:
Strictly speaking we probably don't need `BeginCreateTransactionAsync()` and `BeginDeleteTransactionAsync()` since this will likely only ever be triggered by the `MessageReceived` event from Discord. But, it's the established pattern, it costs basically nothing to implement and test, and I'd much rather have it and not need it, than have a change occur later, and not realize that it's missing and have potential synchronization bugs.